scripts/spack/configs/toss_4_x86_64_ib_cray/spack.yaml

```diff
@@ -0,0 +1,346 @@
+spack:
+  config:
+    install_tree:
+      root: $spack/..
+      projections:
+        all: '{compiler.name}-{compiler.version}/{name}-{version}-{hash}'
+    misc_cache: $spack/../misc_cache
+    test_stage: $spack/../test_stage
+    build_stage::
+    - $spack/../build_stage
+
+  # Regular TPLs do not need views
+  view: false
+
+  compilers::
+  - compiler:
+      environment: {}
+      extra_rpaths: []
+      flags: {}
+      modules: []
+      operating_system: rhel8
+      paths:
+        cc: /opt/rocm-5.7.1/llvm/bin/amdclang
+        cxx: /opt/rocm-5.7.1/llvm/bin/amdclang++
+        f77: /opt/rocm-5.7.1/llvm/bin/amdflang
+        fc: /opt/rocm-5.7.1/llvm/bin/amdflang
+      spec: clang@17.0.0
+      target: x86_64
+
+  packages:
+    all:
+      providers:
+        blas: [netlib-lapack]
+        lapack: [netlib-lapack]
+        gl: [opengl]
+        glu: [openglu]
+        zlib-api: [zlib]
+        mpi: [cray-mpich]
+
+    # Providers
+    # NOTE: Avoid external blas or lapack providers. Compiler was treating shared object path as cpp file in MFEM, which
+    # resulted in "expected unqualified-id" errors.
+    opengl:
+      buildable: false
+      externals:
+      - spec: opengl@1.7.0
+        prefix: /usr
+    openglu:
+      buildable: false
+      externals:
+      - spec: openglu@1.3.1
+        prefix: /usr
+    zlib:
+      buildable: false
+      externals:
+      - spec: zlib@1.2.11
+        prefix: /usr
+
+    # Lock down which MPI we are using
+    mpi:
+      buildable: false
+    cray-mpich:
+      buildable: false
+      externals:
+      - spec: cray-mpich@8.1.27+slurm
+        prefix: /usr/tce/packages/cray-mpich-tce/cray-mpich-8.1.27-rocmcc-5.7.1/
+
+    # ROCm
```

Comment:
To confirm this now matches the hip/llvm subdirectory pattern in Axom  correct? The reasons for this have been explained to me a couple times but leaves my brain immediately, I just know that is the recommended pattern.

Reply:
Yes they match now. Petsc will not be able to build this way, so I set `~petsc` for toss4 cray builds.